unittests/allplatformstests.py

```diff
@@ -3677,7 +3677,7 @@ def test_link_language_linker(self):
         self.init(testdir)
 
         build_ninja = os.path.join(self.builddir, 'build.ninja')
-        with open(build_ninja, encoding='utf-8') as f:
+        with open(build_ninja, encoding=getencoding()) as f:
```

Comment:
This is definitely wrong -- we write it as utf-8, hence why we open it like that, and in general ninja expects:
> Ninja is mostly encoding agnostic, as long as the bytes Ninja cares about (like slashes in paths) are ASCII. This means e.g. UTF-8 or ISO-8859-1 input files ought to work.

Allowing any codepage even if it's not ASCII-compatible at all seems like an error.

Reply:
I changed for reading as bytes.